metrics_listener.go

```diff
@@ -0,0 +1,57 @@
+package sundheitotel
+
+import (
+	"time"
+
+	gosundheit "github.com/AppsFlyer/go-sundheit"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+type MetricsListener struct {
+	classification string
+	gauge          Int64Gauge
+}
+
+func NewMetricsListener(opts ...Option) *MetricsListener {
+	listener := &MetricsListener{}
+
+	for _, opt := range append(opts, WithDefaults()) {
+		opt(listener)
+	}
+
+	return listener
+}
+
+func (c *MetricsListener) OnCheckRegistered(name string, result gosundheit.Result) {
+	c.recordCheck(name, result)
+}
+
+func (c *MetricsListener) OnCheckStarted(_ string) {
+}
+
+func (c *MetricsListener) OnCheckCompleted(name string, result gosundheit.Result) {
+	c.recordCheck(name, result)
+}
+
+func (c *MetricsListener) OnResultsUpdated(results map[string]gosundheit.Result) {
+	allHealthy := allHealthy(results)
+	tags := c.defaultTags(ValAllChecks, allHealthy)
+	mStatus.Record(status(allHealthy).asInt64(), tags...)
+}
+
+func (c *MetricsListener) recordCheck(name string, result gosundheit.Result) {
+	duration := int64(result.Duration) / int64(time.Millisecond)
+	tags := c.defaultTags(name, result.IsHealthy())
+	mDuration.Record(duration, tags...)
```

Comment:
I think you forgot to update the status of the individual check here

Reply:
it's not purpose. We need the status recorded on checkupdated, on registration we don't need to report status yet or on completed. WDYT?